scm/driver/github/webhook.go

```diff
@@ -182,6 +185,74 @@ func (s *webhookService) parsePullRequestHook(data []byte) (scm.Webhook, error)
 	return dst, nil
 }
 
+func (s *webhookService) parsePipelineHook(data []byte) (scm.Webhook, error) {
+	src := new(pipelineHook)
+	err := json.Unmarshal(data, src)
+	if err != nil {
+		return nil, err
+	}
+	dst, err := convertPipelineHook(src), nil
+	return dst, nil
+}
+
+func convertPipelineHook(src *pipelineHook) *scm.PipelineHook {
+	pr := scm.PullRequest{}
+	if len(src.WorkflowRun.PullRequests) > 0 {
+		pr = scm.PullRequest{
+			Number:  src.WorkflowRun.PullRequests[0].Number,
+			Sha:     src.WorkflowRun.PullRequests[0].Head.SHA,
+			Ref:     src.WorkflowRun.PullRequests[0].Head.Ref,
+			Source:  src.WorkflowRun.PullRequests[0].Head.Ref,
+			Target:  src.WorkflowRun.PullRequests[0].Base.Ref,
+			Link:    src.WorkflowRun.PullRequests[0].URL,
+			Created: src.WorkflowRun.CreatedAt,
+		}
+	}
+
+	return &scm.PipelineHook{
+		Repo: scm.Repository{
```

Comment:
can you use the existing `convertRepository` function to handle the conversion logic?

Reply:
updated in latest changes